### PR TITLE
rgw : add check empty for sync url

### PR DESCRIPTION
**Pull request by BryceCao**
Fixes: https://tracker.ceph.com/issues/50103
Signed-off-by: caolei <halei15848934852@163.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_rest_client.cc [hunk 1]
```diff
@@ -813,6 +813,10 @@ int RGWRESTStreamRWRequest::do_send_prepare(RGWAccessKey *key, map<string, strin
                                          bufferlist *send_data)
 {
   string new_url = url;
+  if (new_url.empty() && host_style != VirtualStyle) {
+    ldout(cct, 5) << "ERROR: do_send_prepare has empty target url" << dendl;
+    return -EINVAL;
+  }
```
Comment (cbodley) [inline src/rgw/rgw_rest_client.cc:819]: i don't think we need a new error path, and this doesn't prevent the next line from crashing with VirtualStyle access

can we an empty check to that instead? something like:
```c++
if (!new_url.empty() && new_url.back() != '/')
```

Comment (BryceCao) [reply src/rgw/rgw_rest_client.cc]: > i don't think we need a new error path, and this doesn't prevent the next line from crashing with VirtualStyle access
> 
> can we an empty check to that instead? something like:
> 
> ```c++
> if (!new_url.empty() && new_url.back() != '/')
> ```

THX. i have modifide the code

**Comment by cbodley** [comment]
thanks @BryceCao! do we know where RGWSyncLogTrimThread is sending a request to an empty url? that sounds like a bug too

cc @yuvalif, who also saw this in testing

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
thanks @ivancich!

**Pull request merged**